test/features/step_definitions/DungeonMapSteps.cpp

```diff
@@ -114,6 +135,27 @@ WHEN( "^I call FillArea for a hallway east$" )
     context->map.FillArea( DUNG_IDX_FLOOR, &( context->area ), DIR_EAST, true );
 }
 
+WHEN( "^I call CheckArea$" )
+{
+    ScenarioScope<TestCtx> context;
+    bool areaChecked = context->map.CheckArea( context->dungeonCreationStep );
+    context->areaChecked = areaChecked;
+}
+
+WHEN("^I call get room rect$")
+{
+    ScenarioScope<TestCtx> context;
+    JIVector vPos;
+    vPos.Init(3, 3);
```

Comment:
a "north" room is one that comes from a "northbound" hallway, and is carved from its south wall (from bottom-to-top). 

You are asking for a 10x10 room starting at 3,3 (south wall = 3, north wall = -7 OOB)